### ISSUE TITLE
No GPU support for Windows10 + CUDA 11.0 + Tensorflow 2.4.1: Could not load dynamic library 'cudart64_110.dll'

### ISSUE THREAD

**Issue by mbobinger** [tensorflow/tensorflow#46606]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows10
- TensorFlow installed from (source or binary): pip / packet manager of PyCharm
- TensorFlow version: 2.4.1
- Python version: 3.8
- Installed using virtualenv? pip? conda?: pip / from PyCharm settings menu but it is pip install tensorflow
- CUDA/cuDNN version: CUDA: cuda_11.0.2_451.48_win10 and cuDNN: 8.0.4: cudnn-11.0-windows-x64-v8.0.4.30
- GPU model and memory: GeForce MX330 with 2GB

**Describe the problem**
Hi Guys, I've been trying for around 6hours to get Tensorflow 2.4.1 working with GPU support and following tutorial from pyimagesearch:
https://www.pyimagesearch.com/2018/09/10/keras-tutorial-how-to-get-started-with-keras-deep-learning-and-python/
Using only CPU it worked fine.
I have followed this guide here:
https://www.tensorflow.org/install/gpu
I installed the newest drivers for my GeforceMX330, as well as VisualStudio, CUDA11.0 and the cuDNN 8.0.4 version. Please see my log output

**Any other info / logs**
`train_vgg.py --dataset animals --model output/smallvggnet.model --label-bin output/smallvggnet_lb.pickle --plot output/smallvggnet_plot.png
2021-01-22 15:21:23.627735: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2021-01-22 15:21:23.628081: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
[INFO] loading images...
2021-01-22 15:21:42.850382: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
2021-01-22 15:21:42.852694: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library nvcuda.dll
2021-01-22 15:21:43.635517: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:02:00.0 name: GeForce MX330 computeCapability: 6.1
coreClock: 1.594GHz coreCount: 3 deviceMemorySize: 2.00GiB deviceMemoryBandwidth: 52.21GiB/s
2021-01-22 15:21:43.636419: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2021-01-22 15:21:43.637390: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cublas64_11.dll'; dlerror: cublas64_11.dll not found
2021-01-22 15:21:43.638212: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cublasLt64_11.dll'; dlerror: cublasLt64_11.dll not found
2021-01-22 15:21:43.639402: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cufft64_10.dll'; dlerror: cufft64_10.dll not found
2021-01-22 15:21:43.640209: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'curand64_10.dll'; dlerror: curand64_10.dll not found
2021-01-22 15:21:43.641187: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cusolver64_10.dll'; dlerror: cusolver64_10.dll not found
2021-01-22 15:21:43.642002: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cusparse64_11.dll'; dlerror: cusparse64_11.dll not found
2021-01-22 15:21:43.642798: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'cudnn64_8.dll'; dlerror: cudnn64_8.dll not found
2021-01-22 15:21:43.643052: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1757] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2021-01-22 15:21:43.645778: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-01-22 15:21:43.647564: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-01-22 15:21:43.647835: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      
2021-01-22 15:21:43.648038: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
[INFO] training network...
2021-01-22 15:21:44.544947: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)`

**Comment by amahendrakar** [comment]
@mbobinger,
Please take a look at issues [#43193](https://github.com/tensorflow/tensorflow/issues/43193#issuecomment-749465330) and [#45055](https://github.com/tensorflow/tensorflow/issues/45055#issuecomment-732154795) with a similar error log and check if it works. Thanks!

**Comment by mbobinger** [comment]
@amahendrakar thank you a lot, adding the environment variable for cuda fixed it for me, somehow creating the folder in c:/tools/cuda wasn't 100% clear to me.

**Comment by rednag** [comment]
> @mbobinger,
> Please take a look at issues [#43193](https://github.com/tensorflow/tensorflow/issues/43193#issuecomment-749465330) and [#45055](https://github.com/tensorflow/tensorflow/issues/45055#issuecomment-732154795) with a similar error log and check if it works. Thanks!

Got the same error - can not set the path because it is not available, but CUDA driver 11.2.135 is installed. Can not install other CUDA version, because I've already the newest... so how to fix it?! Uninstall everything?

**Comment by amahendrakar** [comment]
@rednag,
Could you please submit a new issue from [this link](https://github.com/tensorflow/tensorflow/issues/new/choose) and fill in the template, so that we can track the issue there. Thanks!

**Comment by bordeprashant** [comment]
You have either not installed CUDA correctly (and also cudnn) or you have installed the wrong version, for the TF binary you are trying to use.
Install : pip install tensorflow-gpu==2.1
